### DISCUSSION TITLE
PandasAI is good for playing or to experiment when you have too much time

### DISCUSSION THREAD

**Discussion by Manfredzimmermann** [sinaptik-ai/pandas-ai#1334] [General]
I don't want to be rude, but PandasAI is good for playing, but not for a productive environment. I've been working with PandasAI for a week. I've tested SmartDataframe, Agent and etc. with a csv file (titanic.csv). When I ask a question or when I want to chat with my csv file, I sporadically get a correct answer. When I press the st.button ("Generate"), I get the correct answer once, when I ask the same question, it draws me a bar chart and when I ask the same question, I get an error message.

An example:
Prompt: Return the percentage of passengers by gender
Response: The percentage of male passengers is 64.76% and the percentage of female passengers is 35.24%.
Error: 'str' object has no attribute 'get' --> ????

The same thing again: Now a table appears!?!
0 Male 577 99.88776655443323
1 Female 314 99.88776655443323
No code found in the response. --> ????

The same thing again: Now a table appears again!?!
Male 64.75869809203144
% Passengers 35.24130190796858
No code found in the response. --> ????

I really don't feel like playing with it anymore. The results are different every time and these error messages!

**Comment by computersrmyfriends** [comment]
💯 True. I guess it's the same story for a lot of people. search for  "No code found in the response" and "custom prompt" in the discussions and issues. Weirdly, there is no answer for it that actually works although there are answers.

**Comment by Manfredzimmermann** [comment]
Let's assume we have data about patients and their medication intake. It is crucial that we receive a 100% accurate response from PandasAI because a patient's life depends on it. As a doctor, I need to know, for example, how many milligrams of a medication the patient has taken today in order to administer the next dose correctly. If I receive an incorrect response from PandasAI, it could have serious consequences for the patient! When I ask the same question multiple times to PandasAI, I get different answers and even an error message on top of that!
